platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronisedInstallerImpl.java

```diff
@@ -0,0 +1,566 @@
+    }
+    return false;
+  }
+
+  protected boolean wait(Callable<Boolean> conditionIsMet, long maxWait, long pollInterval)
+      throws SynchronisedInstallerException {
+    return wait(conditionIsMet, maxWait, pollInterval, null);
+  }
+
+  private Set<Bundle> getUnavailableBundles(Set<String> toCheck)
+      throws SynchronisedInstallerException {
+    Set<Bundle> unavailableBundles = new HashSet<>();
+    Set<Bundle> failedBundles = new HashSet<>();
+
+    for (Bundle bundle : bundleContext.getBundles()) {
+      if (!toCheck.contains(bundle.getSymbolicName())) {
+        continue;
+      }
+
+      String bundleName = bundle.getHeaders().get(Constants.BUNDLE_NAME);
+      BundleInfo bundleInfo = bundleService.getInfo(bundle);
+      BundleState bundleState = bundleInfo.getState();
+
+      if (BundleState.Failure.equals(bundleState)) {
+        failedBundles.add(bundle);
+      }
+
+      if (bundleInfo.isFragment()) {
+        if (!BundleState.Resolved.equals(bundleState)) {
+          LOGGER.debug("Fragment [{}] not ready with state [{}]", bundleName, bundleState);
+          unavailableBundles.add(bundle);
+        }
+      } else if (!BundleState.Active.equals(bundleState)) {
+        LOGGER.debug("Bundle [{}] not ready with state [{}]", bundleName, bundleState);
+        unavailableBundles.add(bundle);
+      }
+    }
+
+    if (!failedBundles.isEmpty()) {
+      printBundleInformation(LOGGER::error, failedBundles);
+      throw new SynchronisedInstallerException(
+          "Bundles failed to start ["
+              + failedBundles.stream().map(Bundle::getSymbolicName).collect(Collectors.joining(","))
+              + "].");
+    }
+
+    if (!unavailableBundles.isEmpty()) {
+      LOGGER.info(
+          "Waiting on the following bundles to reach Active state [{}]",
+          unavailableBundles
+              .stream()
+              .map(Bundle::getSymbolicName)
+              .collect(Collectors.joining(",")));
+    }
+
+    return unavailableBundles;
+  }
+
+  @VisibleForTesting
+  protected ServiceTracker getServiceTracker(BundleContext bundleContext, Filter filter) {
+    return new ServiceTracker<>(bundleContext, filter, null);
+  }
+
+  private void printBundleInformation(
+      BiConsumer<String, Object[]> logConsumer, Collection<Bundle> bundles) {
+    for (Bundle bundle : bundles) {
+      StringBuilder headerString = new StringBuilder("[ ");
+      Enumeration<String> keys = bundle.getHeaders().keys();
+
+      while (keys.hasMoreElements()) {
+        String key = keys.nextElement();
+        headerString.append(key).append("=").append(bundle.getHeaders().get(key)).append(", ");
+      }
+
+      headerString.append(" ]");
+      logConsumer.accept(
+          "\n\tBundle: {}_v{} | {}\n\tHeaders: {}",
+          new Object[] {
+            bundle.getSymbolicName(),
+            bundle.getVersion(),
+            BUNDLE_STATES.getOrDefault(bundle.getState(), "UNKNOWN"),
```

Comment:
❓ Could you just use the enum's `name()` method to get what you need instead of writing a map? Might look like this assuming nullable since you added an "UNKNOWN" state:
```
bundle.getState() == null ? "UNKNOWN" : bundle.getState().name()
```
Could be a method if it's reused enough but this is the only case I can find.